support/scripts/install

```diff
@@ -1,38 +1,259 @@
+choose_wget_or_curl() {
 
-  echo "Verifying checksum..."
-  curl -sSL "https://github.com/sundowndev/phoneinfoga/releases/download/$phoneinfoga_version/phoneinfoga_checksums.txt" -o phoneinfoga_SHA256SUMS
-  sha256sum --ignore-missing -c phoneinfoga_SHA256SUMS
-  [ $? -eq 0 ] || exit 1
+    which curl > /dev/null
+    [ $? -eq 1 ] || CURL_INSTALLED=true
 
-  tar xfv "phoneinfoga_$os.tar.gz"
-  [ $? -eq 0 ] || exit 1
-  
-  # Clear downloaded assets
-  rm phoneinfoga_$os.tar.gz
-  rm phoneinfoga_SHA256SUMS
+    if $CURL_INSTALLED; then
+        PHONEINFOGA_VERSION=$(curl -s https://api.github.com/repos/sundowndev/phoneinfoga/releases/latest | grep tag_name | cut -d '"' -f 4)
+        return
+    fi
 
-  echo "Installation completed successfully."
-  echo "Check the version : ./phoneinfoga version"
-  echo "You can now install the program globally : sudo mv ./phoneinfoga /usr/bin/phoneinfoga"
-else
-  echo "Your OS/Arch is not supported."
-  echo "Read more at https://sundowndev.github.io/phoneinfoga/install/"
-  exit 1
-fi
+    which wget > /dev/null
+    [ $? -eq 1 ] || WGET_INSTALLED=true
 
-exit 0
+    if $WGET_INSTALLED; then
+        PHONEINFOGA_VERSION=$(wget -q --output-document - https://api.github.com/repos/sundowndev/phoneinfoga/releases/latest | grep tag_name | cut -d '"' -f 4)
+        return
+    fi
 
+    echo "Error: You need to have either curl or wget installed to be able to use this script"
+    exit 1
+
+}
+
+
+usage() {
+
+    echo "PhoneInfoga Installer for version $PHONEINFOGA_VERSION"
+    echo
+    echo "DESCRIPTION: An installer script for downloading the latest release of PhoneInfoga. Without any arguments, $0 will detect your operating system and attempt to download the corresponding version of PhoneInfoga. Please submit an issue on GitHub if you encounter issues with this script." | fold -s
+    echo
+    echo "USAGE: $0 [flag...] (-h|-m|-s)"
+    echo
+    echo "      -h | --help             Print this message and exit"
+    echo
+    echo "      -m | --manual           Manually select version to download."
+    echo "                              Useful when $0 is unable to detect"
+    echo "                              your OS automatically"
+    echo
+    echo "      -s | --skip-checksum    [Not Reccomended] Skip checksum validation"
+    echo "                              Only use this option if $0"
+    echo "                              is failing due to missing software."
+    echo
+}
+
+
+validate_OS_type() {
+
+    if [[ -z "${OS}" ]]; then
+        echo "ERROR: Unable to determine your system type."
+        exit 1
+    fi
+}
+
+
+validate_supported_OS() {
+
+    SUPPORTED_OS_TYPES=(
+        "Darwin_arm64"
+        "Darwin_x86_64"
+        "Linux_arm64"
+        "Linux_armv6"
+        "Linux_armv7"
+        "Linux_i386"
+        "Linux_x86_64"
+        "Windows_arm64"
+        "Windows_armv6"
+        "Windows_armv7"
+        "Windows_i386"
+        "Windows_x86_64"
```

Comment:
You may have guessed those targets but they actually don't exist in releases. On a side note, since this script is using jq, I think it'd be cool to fetch that list from the github api so it's always up to date. WDYT?